pygmt/base_plotting.py

```diff
@@ -55,1592 +40,17 @@ def _preprocess(self, **kwargs):  # pylint: disable=no-self-use
-
-        # Build the `-F` argument in gmt text.
-        if "F" not in kwargs.keys() and (
-            (
-                position is not None
-                or angle is not None
-                or font is not None
-                or justify is not None
-            )
-        ):
-            kwargs.update({"F": ""})
-        if angle is not None and isinstance(angle, (int, float, str)):
-            kwargs["F"] += f"+a{str(angle)}"
-        if font is not None and isinstance(font, str):
-            kwargs["F"] += f"+f{font}"
-        if justify is not None and isinstance(justify, str):
-            kwargs["F"] += f"+j{justify}"
-        if position is not None and isinstance(position, str):
-            kwargs["F"] += f'+c{position}+t"{text}"'
-
-        extra_arrays = []
-        # If an array of transparency is given, GMT will read it from
-        # the last numerical column per data record.
-        if "t" in kwargs and is_nonstr_iter(kwargs["t"]):
-            extra_arrays.append(kwargs["t"])
-            kwargs["t"] = ""
-
-        with Session() as lib:
-            file_context = dummy_context(textfiles) if kind == "file" else ""
-            if kind == "vectors":
-                if position is not None:
-                    file_context = dummy_context("")
-                else:
-                    file_context = lib.virtualfile_from_vectors(
-                        np.atleast_1d(x),
-                        np.atleast_1d(y),
-                        *extra_arrays,
-                        # text must be in str type, see issue #706
-                        np.atleast_1d(text).astype(str),
-                    )
-            with file_context as fname:
-                arg_str = " ".join([fname, build_arg_string(kwargs)])
-                lib.call_module("text", arg_str)
-
-    # GMT Supplementary modules
+    from pygmt.src import basemap  # pylint: disable=import-outside-toplevel
+    from pygmt.src import coast  # pylint: disable=import-outside-toplevel
+    from pygmt.src import colorbar  # pylint: disable=import-outside-toplevel
+    from pygmt.src import contour  # pylint: disable=import-outside-toplevel
+    from pygmt.src import grdcontour  # pylint: disable=import-outside-toplevel
+    from pygmt.src import grdimage  # pylint: disable=import-outside-toplevel
+    from pygmt.src import grdview  # pylint: disable=import-outside-toplevel
+    from pygmt.src import image  # pylint: disable=import-outside-toplevel
+    from pygmt.src import legend  # pylint: disable=import-outside-toplevel
+    from pygmt.src import logo  # pylint: disable=import-outside-toplevel
     from pygmt.src import meca  # pylint: disable=import-outside-toplevel
+    from pygmt.src import plot  # pylint: disable=import-outside-toplevel
+    from pygmt.src import plot3d  # pylint: disable=import-outside-toplevel
+    from pygmt.src import text  # pylint: disable=import-outside-toplevel
```

Comment:
You know, if we're going to move **every** plotting function to the `src/` folder, we might as well delete `base_plotting.py` and import everything in `figure.py` instead. Or we could keep a **few** core functions in `base_plotting.py` (e.g. `basemap`, and `inset` perhaps as mentioned in https://github.com/GenericMappingTools/pygmt/pull/788#issuecomment-766760050). What do people think?

```suggestion
from pygmt.src import (  # pylint: disable=import-outside-toplevel
        basemap,
        coast,
        colorbar,
        contour,
        grdcontour,
        grdimage,
        grdview,
        image,
        legend,
        logo,
        meca,
        plot,
        plot3d,
        text,
    )
```

Reply:
> You know, if we're going to move **every** plotting function to the `src/` folder, we might as well delete `base_plotting.py` and import everything in `figure.py` instead. 

I like this one.

Reply:
I committed this changed in 76c4969 but it led to 191 tests failing when I ran `make test`. I'm not sure what the issue is, but I've reverted for now.

Reply:
Oops sorry, just realized it was missing an indentation... Try this instead:

```suggestion
    from pygmt.src import (  # pylint: disable=import-outside-toplevel
        basemap,
        coast,
        colorbar,
        contour,
        grdcontour,
        grdimage,
        grdview,
        image,
        legend,
        logo,
        meca,
        plot,
        plot3d,
        text,
    )
```

Reply:
Ok, I applied https://github.com/GenericMappingTools/pygmt/pull/808#discussion_r570775763 in 127b42874466c66eb970fb53a8846db9ff1f73e8.